src/ice/candidate.rs

```diff
@@ -117,6 +122,51 @@ impl fmt::Debug for Candidate {
 }
 
 impl Candidate {
+    /// Starts the typesafe builder.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use str0m::Candidate;
+    /// # use str0m::error::IceError;
+    /// # use str0m::net::{Protocol, TcpType};
+    /// # use std::net::SocketAddr;
+    ///
+    /// let addr: SocketAddr = "192.168.1.1:12345".parse().unwrap();
+    ///
+    /// // A standard UDP Host candidate
+    /// let udp_host = Candidate::builder()
+    ///     .udp()
+    ///     .host(addr)
+    ///     .build()?;
+    ///
+    /// // A TCP Host candidate with a passive role
+    /// let tcp_host = Candidate::builder()
+    ///     .tcp()
+    ///     .host(addr)
+    ///     .tcptype(TcpType::Passive)
```

Comment:
Would it be worth reducing the public API surface and changing this to:

```
.passive()
```

`TcpType` could then be a crate-private type.